exercises/practice/sieve/sieve.spec.js

```diff
@@ -0,0 +1,76 @@
+import { compileWat, WasmRunner } from "@exercism/wasm-lib";
+
+let wasmModule;
+let currentInstance;
+
+beforeAll(async () => {
+  try {
+    const watPath = new URL("./sieve.wat", import.meta.url);
+    const { buffer } = await compileWat(watPath);
+    wasmModule = await WebAssembly.compile(buffer);
+  } catch (err) {
+    console.log(`Error compiling *.wat: \n${err}`);
+    process.exit(1);
+  }
+});
+
+function primes(limit) {
+  const [outputOffset, outputLength] = currentInstance.exports.primes(limit);
+
+  const outputBuffer = currentInstance.get_mem_as_i32(
+    outputOffset,
+    outputLength
+  );
+
+  return [...outputBuffer];
```

Comment:
I don't quite understand why this is required. Does the JS side assume the chunk of memory is a string, and this coerces it to an array of numbers?

Reply:
It's because the interfaces with linear memory return JavaScript `TypedArrays`. 

In this case, you're getting an `Int32Array`. The error message `Received: serializes to the same string` seems to suggest that both the JS array and the `Int32Array` serialize to identical strings. True, but not particularly helpful.

Your solution here is fine. It converts the `Int32Array` to a JS array. If you wanted to keep things typed, you could also refactor your tests to compare against an `Int32Array` by wrapping your JS arrays in a call to the `Int32Array` ctor:

```js
  test("find primes up to 10", () => {
    expect(primes(10)).toEqual(new Int32Array([2, 3, 5, 7]));
  });
```
  
Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Int32Array

I'm fine with either approach.